docker/models/services.py

```diff
@@ -0,0 +1,89 @@
+import inspect
+from docker.types import TaskTemplate, ContainerSpec
+from .resource import Model, Collection
+
+
+class Service(Model):
+    """A service."""
+    id_attribute = 'ID'
+
+    @property
+    def name(self):
+        """The service's name."""
+        return self.attrs['Spec']['Name']
+
+    @property
+    def version(self):
+        """The version number of this service."""
+        return self.attrs.get('Version').get('Index')
+
+    def remove(self):
+        """Remove this service."""
+        return self.client.api.remove_service(self.id)
+
+    def tasks(self):
+        """List the tasks in this service."""
+        return self.client.api.tasks(filters={'service': self.id})
+
+    def update(self, **kwargs):
+        """Update a service's configuration."""
+        create_kwargs, spec_kwargs = _get_create_and_spec_kwargs(kwargs)
+        # Image is required, so if it hasn't been set, use current image
+        if 'image' not in spec_kwargs:
+            spec = self.attrs['Spec']['TaskTemplate']['ContainerSpec']
+            spec_kwargs['image'] = spec['Image']
+        return self.client.api.update_service(
+            self.id,
+            self.version,
+            TaskTemplate(ContainerSpec(**spec_kwargs)),
+            **create_kwargs
+        )
+
+
+class ServiceCollection(Collection):
+    """Services on the Docker server."""
+    model = Service
+
+    def create(self, **kwargs):
+        """Create a service."""
+        create_kwargs, spec_kwargs = _get_create_and_spec_kwargs(kwargs)
+        service_id = self.client.api.create_service(
+            TaskTemplate(ContainerSpec(**spec_kwargs)),
+            **create_kwargs
+        )
```

Comment:
This is how it currently works, [from the docs](https://docker-py.readthedocs.io/en/stable/services/):

```python
container_spec = docker.types.ContainerSpec(
    image='busybox', command=['echo', 'hello']
)
task_tmpl = docker.types.TaskTemplate(container_spec)
service_id = client.create_service(task_tmpl, name=name)
```

I want it to be more like this:

```python
service = client.services.create(image='busybox', command=['echo', 'hello'])
```

If you want me to explain why that is better for the user, I can, but hopefully it's self-explanatory. ;)